baldm0mma/addTableNums/ adding table row numbers to table panel

baldm0mma opened the pull request:
Draft PR - adding table row numbers to table panel

Jguer committed:
SupportBundles: Fix collector typos and change defaults (#61850)

fix collector typos and change defaults

baldm0mma committed:
baldm0mma/addTableNums/ initial commit

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=61873) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=61873) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana?pullRequest=61873) it.</sub>

kylebrandt committed:
SSE/Prom: Undo empty frame change that triggers 500s in SSE (#61871)

added in https://github.com/grafana/grafana/pull/61643

MattDodsonEnglish committed:
Docs: rewrite data source intro (#61639)

* Docs: rewrite data source intro

- Include more than just "databases", which was overly narrow
- Try to make doc a "TL;DR" of the essential information

* docs, link data source to build plugin task

* Reduce jargon and sentence length

* Link out to different features

* Data source, docs turn examples into list

And trim language

* Better phrasing to introduce query editor

Old way implied that Grafana provides all query editors,
which isn't true of custom plugins.

* use relref for in-repo anchor

* remove double "in"

Co-authored-by: Eve Meelan <81647476+Eve832@users.noreply.github.com>

* Simplify list

Co-authored-by: Eve Meelan <81647476+Eve832@users.noreply.github.com>

* Restructure intro, clarify plugins and their URLs

* Add and configure

It's sometimes both.

* Apply suggestions from code review

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

* makes prettier

Co-authored-by: Eve Meelan <81647476+Eve832@users.noreply.github.com>
Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>
Co-authored-by: Chris Moyer <chris.moyer@grafana.com>

mildwonkey committed:
chore: move jwt models into auth/jwt (#61862)

* chore: move jwt models into auth/jwt

asimpson committed:
Chore: update @grafana/experimental to 1.1.0 (#61820)

The main goal is to bring in changes that address a11y issue #56561.

This also fixes any ts errors.

adela-almasan committed:
Canvas: Rectangle render error (#61883)

baldm0mma committed:
jev/addRowNums/ remove console.log

ryantxu committed:
EntityAPI: Save nested summary info in the SQL database (#61732)

nmarrs committed:
Canvas: Update readme with link to docs (#61812)

undef1nd committed:
Codegen: Generate docs for composable kinds (#61719)

* Generate docs for composable kinds

* Fix getting the kind schema

* Generate composable kinds docs

* Add disclaimer to the generated docs

* Change docs disclaimer

ivanortegaalba committed:
test: use deterministic timestamp to compare snapshots (#61670)

jackw committed:
Chore: Bump Terser related dependencies (#60506)

chore(yarn): refresh lock file for all dependencies of terser

mdvictor committed:
Docs: Update visualization and barchart panels (#61474)

* Update docs

* Regen cue model for barchart

* Fix docs

* models.cue renamed to composable_panelcfg.cue

* Rename 'Bar' to 'x-axis' in options UI

* run gen-cue

jackw committed:
Chore: Patch bump QS  (#60508)

chore(yarn): patch bump qs to resolve dependabot alerts

ashharrison90 committed:
Navigation: clean up colors in section nav (#61240)

* clean up colors in section nav

* just underline on hover/focus

ashharrison90 committed:
Navigation: Change collapse icon + behaviour (#61496)

* adjust positioning and switch icon

* hide collapse button unless collapsed/focused

* use Button instead of IconButton

kalleep committed:
RBAC: Rewrite data source api tests (#61783)

RBAC: Rewrite datasource rbac api tests

torkelo committed:
Apps: Use plugin description as nav node subtitle (#61549)

torkelo committed:
TopNav: Fixes breadcrumb issues and title for apps when topnav is disabled (#61835)

gotjosh committed:
Alerting: Remove unused code after importing from grafana/alerting (#61869)

* Alerting: Remove unused code after importing from grafana/alerting

zerok committed:
CI: Add `artifacts npm` commands from grabpl (#61908)

axelavargas committed:
PanelChrome: Allow panel to be dragged if set as draggable from the outside (#61698)

kalleep committed:
AuthN: Add interface and function to operate on clients that supports redirects (#61905)

wbrowne committed:
Plugins: Add file permission error check when attempting to verify plugin signature (#61860)

add permission err check

jackw committed:
Build: Replace babel-loader with esbuild-loader (#57837)

* build(webpack): replace babel-loader with esbuild-loader

* build(webpack): add esbuild minifier to production builds

* Wip

* Removed ngInject and replaced with manual inject params

* chore: bump esbuild to 0.15.13

* Fixed angular issues

* build(frontend): update esbuild to 0.16.16

* chore(webpack): support browserslist for esbuild

* build(esbuild): unify versions of esbuild to 0.16.17 and esbuild-loader to 2.21.0

Co-authored-by: Torkel Ödegaard <torkel@grafana.com>

gabor committed:
elastic: improve error-messages, first step, tests only (#61847)

papagian committed:
Chore: Drop search service dependency from folder service (#61789)

* Chore: Drop search service dependency from folder service

aangelisc committed:
AzureMonitor: Move DefaultSubscriptions field to separate component (#61633)

* Move DefaultSubscriptions field to separate component

- Add separate component
- Update CredentialsForm
- Update tests
- Remove redundant test

* Remove defaultSubscriptionId

- Update MonitorConfig to appropriately set subscriptionId
- Update DefaultSubscription component to make use of JsonData
- Remove references to defaultSubscriptionId
- Update tests
- Update mockInstanceSettings for flexibility

* Review

- Add DefaultSubscription test and remove component from AzureCredentialsForm test
- Remove onCredentialsChange from DefaultSubscription
- Set subscription as a part of onSubscriptionChange

leventebalogh committed:
Connections: Show the Dashboards tab under DataSource settings (#61841)

* fix: use the correct URL for editing datasource dashboards

* feat: show the Datasource/Dashboards page under Connections as well

* refactor: remove unnecessary `useNavModel()` utility

* feat: add a utility for creating the nav-model for DS settings tabs

* refactor: use nav-model crating utility in Edit Datasource

joanlopez committed:
SQLStore: Add test for nested transactions events (#60500)

* SQLStore: Add test for nested transactions events

* Replace fmt.Print* with t.Log*

* Add different test cases

mildwonkey committed:
chore: move alert-related models (#61716)

* chore: move alert notification models into the alerting service (alerting/models)

Commit:
Kindsys: Generate plugin Go types under a kinds/ dir (#61899)

Kindsys: Generate plugin Go types under kinds/

spinillos committed:
Chore: Remove DecoderCompactor() (#61706)

* Remove ApplyFunc and set NoOptionalPointers to true

* Update go mod

Commit:
Kindsys: Remove reference to DecoderCompactor (#61922)

zerok committed:
CI: Remove `grabpl integration-tests` (#61920)

These tests can also be run without it.

itsgareth committed:
Loki: Add ingester to the devenv loki config (#61844)

feat: update loki config

joanlopez committed:
Loki: Push support for multi-tenancy mode (#60866)

* Loki: Add multitenancy support for the HTTP client

* Loki: Support to push with multitenancy mode

* Apply feedback suggestions

Commit:
Tempo: [TraceQL] Integrate the new tag values API (#61637)

* Tempo: [TraceQL] Integrate the new tag values API

* Use the exact string entered in the spanset

* Update a test & add a new test

* Empty commit

svennergr committed:
Elasticsearch: Fix failing requests when using SigV4 (#61923)

fix setting of `es` servicename

Jguer committed:
SupportBundles: Access control guards (#61914)

* rename routes and fix access control for support bundles

* AccessControl: Hide menu if not authorized

* AccessControl: Add AC guards for create and delete

* lint

Jguer committed:
SAML: Update deps (#61906)

peakle committed:
Perfomance: Add preallocation for some slices (#61632)

adela-almasan committed:
Canvas: Connection heads not rendering consistently (#61887)

mildwonkey committed:
chore: move models/licensing into licensing service (#61878)

joe-elliott committed:
Tempo Datasource: Correct TraceQL docs link (#61931)

* correct link

Signed-off-by: Joe Elliott <number101010@gmail.com>

* lint

Signed-off-by: Joe Elliott <number101010@gmail.com>

Signed-off-by: Joe Elliott <number101010@gmail.com>

ashharrison90 committed:
Navigation: truncate landing page descriptions to 3 lines (#61925)

* truncate landing page descriptions to 3 lines

* use correct css prop names

juanicabanas committed:
GrafanaUI: Checkbox description fix (#61929)

Commit:
Kindsys: Unique names for composable kind TS types (#61928)

* Kindsys: Unique names for composable kind TS types

* Update all TS imports

Commit:
Kindsys: Unique names for composable kind TS types (#61928)

* Kindsys: Unique names for composable kind TS types

* Update all TS imports

Commit:
Kindsys: Unique names for composable kind TS types (#61928)

* Kindsys: Unique names for composable kind TS types

* Update all TS imports

baldm0mma committed:
jev/tableRowNums/ prototype option 1

mildwonkey committed:
chore: move plugins models into pluginsettings svc (#61944)

baldm0mma committed:
jev/tableRowNums/ start prototpe

Commit:
Kindsys: Add Ptr func (#61948)

Commit:
panels: Remove redundant import package from cue defs (#61949)

Commit:
panels: Remove redundant import package from cue defs (#61949)

Commit:
panels: Remove redundant import package from cue defs (#61949)

mildwonkey committed:
chore: move validations model into the validations service (#61953)

juanicabanas committed:
PublicDashboards: Checkboxes list refactor (#61947)

juanicabanas committed:
PublicDashboards: Footer position fix (#61954)

jeremybanzhaf committed:
Docs: Update index.md to not escape dollar sign (#61694)

Update index.md to not escape dollar sign

the backslash is useful in json files, but not where most users will see it

alexweav committed:
Alerting: Push state history entries to Loki (#61724)

* Implement push endpoint

* Drop duplicated struct

* Genericize auth/tenant headers and improve logging in error case

* Flesh out the data model

* Drop dead code

* Drop log line entirely

* Drop unused arg

* Rename a few type manipulation functions

* Extract label keys as constants

* Improve logs when loki responds with error

* Inline lokiRepresentation function

Clarity-89 committed:
Geomap panel: Generate types (#61636)

bohandley committed:
Graphite: Have metric name type query variable use metric render endpoint (#61730)

* have metric name query var use metric render endpoint

* add documentation for new query types

adela-almasan committed:
Canvas: Improve arrow positioning when border is present (#61961)

spinillos committed:
Chore: Add cuefix hook (#61941)

* Add cue

* Add fix-cue target

* Add pre-commit hook

* fixup! Add fix-cue target

* Update the cue version

* Add datasources and panels folders

* Update cue files

Co-authored-by: Tania B <yalyna.ts@gmail.com>

joanlopez committed:
Kindsys: Include generated-code links to Kinds report (#61910)

* Kindsys: Include generated-code links to Kinds report

* Apply feedback suggestions

tolzhabayev committed:
Linking criteria and adding file and env variables access (#61830)

* Linking criteria and adding file and env variables access

* Formatting

* Update docs/sources/developers/plugins/publish-a-plugin.md

Co-authored-by: Marcus Efraimsson <marcus.efraimsson@gmail.com>

Co-authored-by: Marcus Efraimsson <marcus.efraimsson@gmail.com>

soniaAguilarPeiron committed:
Alerting: Add maxdatapoints in alert rule form (#61904)

Add maxdatapoints in alert rule form

papagian committed:
Chore: Use same JSON tag casing everywhere for parent UID (#61935)

Chore: Use same JSON tag casing everywhere for parent UID

torkelo committed:
Profile: Rename profile page from preferences to profile (#61777)

* Rename preferences to Profile

* Fixed tests and ran i18 extract

torkelo committed:
Home: Fixes breadcrumb for custom home dashboard (#61499)

* Home: Fixes breadcrumb for custom home dashboard

* restore using home page cfg setting

joanlopez committed:
Kindsys: Include @grafanamaturity counts to Kinds report (#61911)

* Kindsys: Include @grafanamaturity counts to Kinds report

* Replace reflect.DeepEqual with regular (in)equality op

* Move reference check to its use (inline)

* Fix linter complains

ashharrison90 committed:
Chore: convert last test to RTL and remove Enzyme references (#61918)

convert last test to RTL and remove enzyme references

zoltanbedi committed:
Panel edit: Add feature to drag & drop spreadsheet files to the grafana datasource (#60586)

Co-authored-by: Oscar Kilhed <oscar.kilhed@grafana.com>
Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>
Co-authored-by: Adela Almasan <adela.almasan@grafana.com>

adrapereira committed:
Chore: Update the tempo devenv with the latest Tempo config changes (#61622)

* Update the tempo devenv with the latest Tempo config changes

* Remove some settings to instead use the default values

* Pin the Tempo image to the current latest version

Co-authored-by: Hamas Shafiq <hamas.shafiq@grafana.com>

oscarkilhed committed:
Table panel: Use link elements instead of div elements with on click events to aid with keyboard accessibility (#59393)

* TablePanel: fix image of image cell overflowing table cell when a data link is added

Clarity-89 committed:
AlertGroups: Generate models.gen.ts from models.cue (#61227)

* AlertGroups: Generate models.gen.ts from models.cue

* Update structure

* Update report

* Fix filenames

* Add missing file

ying-jeanne committed:
[xorm] Clean up xorm dialect & cascade (#61969)

clean up xorm dialect

joshhunt committed:
Navigation: Open command palette from search box (#61667)

* Reduce size of topnav search 'input' to 1/5th of the width, min width 200px

* Open command palette on topnav search box click

* Rename component

* fix comment

* feature flag the change

* update feature flag description

andresmgot committed:
Add debug option for Golang tests in vscode (#61983)

iwysiu committed:
CloudWatch: Prevent log groups from being removed on query change (#61891)

chri2547 committed:
Docs: prose and format updates (#61720)

* wording and format updates

* corrects typo

andresmgot committed:
Azure Monitor: Adapt Advanced component to multiple resources (#61981)

joshhunt committed:
CommandPalette: Fix feature flag (#62000)

joanlopez committed:
Kindsys: Include CODEOWNERS to Kinds report (#61978)

* Kindsys: Include CODEOWNERS to Kinds report

* Sort CODEOWNERS for the sake of reproducibility

* Fix linter complains

wbrowne committed:
Plugins: Add backend target to instrumentation (#61980)

* add target

* fix test

leventebalogh committed:
Connections: update the style of the cards (#61989)

grobinson-grafana committed:
Alerting: Return chan <-error for #61811 (#61858)

csmarchbanks committed:
Datasources: Fix Proxy by UID Failing for UIDs with a Hyphen (#61723)

Fix Proxy by UID Failing for UIDs with a Hyphen

Hyphens are allowed in short IDs but not picked up by the
proxyPathRegexp. This caused the end of the uid to be proxied as part of
the path to the backing datasource which would usually cause a 404.

Clarity-89 committed:
Schema: Move known shared types into the common (#61875)

* Setup shared types

* Expand field types

* Fix array type

* Add QueryResultMeta

* Add QueryResultBase

* Add DataFrameJSON

* Add DataSourcePluginMeta

* Update QueryResultMetaStat

* Add plugin cue and veneers

* Rexport dupliate types to fix errors

* Fix export

* Update types

* Remove data frame/plugin types

* Cleanup types

* Remove redundant type

* Remove data frame related types

undef1nd committed:
Docs: Fix schema attributes formatting in schema docs (#62016)

Docs: Fix schema attributes formatting

gelicia committed:
Explore: Be sure time range key bindings are mounted after clear (#61892)

* Add list of bindings for updating on change

* Revert "Add list of bindings for updating on change"

This reverts commit 092707371d0c387c3ee9522b77bef3a72c62797f.

* Clear keybindings before component render

baldm0mma committed:
baldm0mma/addRowNums/ add control flow for hasRowNumberColumn

santihernandezc committed:
Chore: Fix random indices for slices in test files (#61884)

* Fix random indices for slices in test files

* Empty commit

ivanahuckova committed:
Explore: Implement logs sample in Explore (#61864)

* Implement log samples

* Explore: Implement logs sample panel

* Log samples: Add documentation

* Update docs

* Add info for log sample

* Fix label

* Update

* Default to true

* Fix copy in test

* Update public/app/features/explore/LogsSamplePanel.tsx

Co-authored-by: Sven Grossmann <sven.grossmann@grafana.com>

* Use timeZone from grafana/schema

* Rename data props to queryResponse

* Unify name to logs sample

* Remove redundant optional parameters in LogsSamplePanel

* Make intervalMs parameter optional in dataFrameToLogsModel and remove undefined argument when not needed

* Fix incorrect position of copy log line button

* Update public/app/core/logsModel.ts

Co-authored-by: Sven Grossmann <sven.grossmann@grafana.com>

Co-authored-by: Sven Grossmann <sven.grossmann@grafana.com>

baldm0mma committed:
baldm0mma/addRowNums/ add empty column

Commit:
Tempo: [TraceQL] Do not override the `status` tag name (#62030)

baldm0mma committed:
baldm0mma/addRowNums/ remove build inside of getColumns

Commit:
dashboards: Mark dashboards team as kind owner (#62013)

* dashboards: Mark dashboards team as kind owner

* Update .github/CODEOWNERS

Co-authored-by: Ivan Ortega Alba <ivanortegaalba@gmail.com>

Co-authored-by: Ivan Ortega Alba <ivanortegaalba@gmail.com>

chri2547 committed:
Docs: corrects broken link to v9.0 docs (#62035)

corrects broken link to v9.0 docs

aashish0909 committed:
Explore: Changed references to DataQuery and DataSourceRef (#62034)

baldm0mma committed:
balm0mma/addRowNums/ add ROW_NUMBER_WIDTH constant

baldm0mma committed:
balm0mma/addRowNums/ add annos in schema cue for reference

baldm0mma committed:
balm0mma/addRowNums/ add annos in schema cue for reference

baldm0mma committed:
balm0mma/addRowNums/ add annos in schema cue for reference

owensmallwood committed:
PublicDashboards: Adds middleware for email sharing (#61950)

adds Share field to PublicDashboard model

adela-almasan committed:
Canvas: Update scene and panel when entering panel view mode (#62043)

idafurjes committed:
Chore: Remove folders from models pkg (#61853)

academo committed:
Docs: Update publish a plugin FAQ to cover more common questions (#62001)

fridgepoet committed:
CloudWatch: Use grafana-aws-sdk v0.12.0 (#62007)

soniaAguilarPeiron committed:
Alerting: Fix recording rules showing alert notification information. (#61988)

* Fix recording rules form steps not showing alert information

* Fix docs about creating cloud and recording rules

* Update docs with suggested changes

grafanabot committed:
Changelog: Updated changelog for 9.2.10 (#62066)

* Changelog: Updated changelog for 9.2.10

* Update CHANGELOG.md

* Update CHANGELOG.md

Co-authored-by: Horst Gutmann <horst.gutmann@grafana.com>

grafanabot committed:
Changelog: Updated changelog for 9.3.4 (#62067)

idafurjes committed:
Chore: Remove Result from dashboard models (#61997)

* Chore: Remove Result from dashboard models

* Fix lint tests

* Fix dashboard service tests

* Fix API tests

* Remove commented out code

* Chore: Merge main - cleanup

torkelo committed:
CommandPalette: Design tweaks and design fixes  (#61971)

* Command palette design tweaks start

* Fix some item styling issues

* Updated placeholder

* Style fixes

* Fix gradient border pos issue, and heading padding

* Fix header top margin

* Restore padding

* Update

* Change to md modal

grafanabot committed:
Changelog: Updated changelog for 8.5.20 (#62073)

dsotirakis committed:
Chore: update latest.json to 9.3.4 (#62080)

gotjosh committed:
Update alerting to the latest main (#62003)

wbrowne committed:
Plugins: Add plugin resource tests (#62014)

* remove plugin context from response

* remove integration test indicator

renovate[bot] committed:
Update dependency eslint-plugin-jest to v27.2.1 (#61523)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

dprokop committed:
Dashboard schema: Review and mature graphTooltip property  (#62082)

* Review and mature graphTooltip property of Dashboard kind

* Review

eleijonmarck committed:
Auth: Add skip_org_role_sync to GitLab OAuth (#62055)

* Auth: Add skip_org_role_sync to GitLab OAuth

- add: tests
- docs added

* Update pkg/login/social/gitlab_oauth.go

Co-authored-by: Karl Persson <kalle.persson@grafana.com>

* fix: for import

Co-authored-by: Karl Persson <kalle.persson@grafana.com>

renovate[bot] committed:
Update dependency eslint to v8.32.0 (#61894)

* Update dependency eslint to v8.32.0

* update sdk

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Ashley Harrison <ashley.harrison@grafana.com>

sd2k committed:
API: Correctly use new grafana_com.api_url setting in /api/gnet proxy (#60893)

During the review of the initial PR adding this (#59506) I removed
a new global variable from the setting package, but forgot to update
the reference to the new setting, so the API URL wasn't actually
being used. This PR updates the proxy endpoint to use the API
URL correctly.

Aside: I'm not a huge fan of how the error is being ignored when parsing
the URL, but I think that should be addressed in a separate PR if anyone
has a suggestion for how we should handle it. (Should we check that the
URL is valid when parsing config?)

axelavargas committed:
Dashboard schema: Review and mature fiscalYearStartMonth property (#62105)

tolzhabayev committed:
Chore: Keeping the list of plugin executable complete (#60840)

* Keeping the list of executable complete

* Update pkg/plugins/storage/fs.go

Co-authored-by: Will Browne <wbrowne@users.noreply.github.com>

Co-authored-by: Will Browne <wbrowne@users.noreply.github.com>

kylebrandt committed:
API: Change how Cache-Control and related headers are set (#62021)

- change Cache-Control from no-cache to no-store
- do not set (and remove if set) older Pragma/Expires

idafurjes committed:
Chore: Remove Result field from dashboard snapshot mode (#62089)

Chore: Remove Result field from dashboard snapshot mode;

eleijonmarck committed:
Auth: Add skip_org_role_sync setting for github (#61673)

* add: skip_org_role_sync setting for github

* fix: frontend

* rearranged tests

* refactor: assignGrafanaAdmin skip also

* Add: tests for allowGrafanaAdmin

- both for the case when both settings are set and the setting for only
  allowGrafanaAdmin

* Apply suggestions from code review

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

* Update docs/sources/setup-grafana/configure-grafana/_index.md

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

* Update pkg/login/social/github_oauth.go

Co-authored-by: Ieva <ieva.vasiljeva@grafana.com>

* added vairable inside scope

* Update docs/sources/setup-grafana/configure-security/configure-authentication/github/index.md

* Update docs/sources/setup-grafana/configure-security/configure-authentication/github/index.md

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>
Co-authored-by: Ieva <ieva.vasiljeva@grafana.com>

juanicabanas committed:
PublicDashboards: Footer alignment fix for Firefox browser (#62108)

footer x axis aligment fixed

iwysiu committed:
CloudWatch Logs: Set default logs query and disable button when empty (#61956)

ying-jeanne committed:
[API Split] Move star api inside of packages (#61987)

move star api inside of packages

mildwonkey committed:
chore: update folder model (json tags) to match previous model (#62117)

chore: update folder model to match previous model

DanCech committed:
copy .github folder into golang build container since we rely on codeowners (#62122)

* copy .github folder into golang build container since we rely on codeowners

* remove .github for .dockerignore

stephaniehingtgen committed:
Plugins: add UI for more supported datasources with secure socks proxy feature toggle (#61962)

yuri-tceretian committed:
Docs: Update expression documentation to mention no data (#61934)

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

renovate[bot] committed:
Update dependency fork-ts-checker-webpack-plugin to v7.3.0 (#62129)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

polibb committed:
[Chore]  Add unit tests to PanelChrome component (#61695)

* PanelChrome: test loadingState and status

* add some tests

* fix error-related tests

* clean up

* fix tests

* pass aria-label from PanelChrome to ToolbarButton

* add prop comments and clean up

ktw4071 committed:
add export customHeadersSettings component (#62131)

spinillos committed:
Schema: Fix interfaces with blank spaces (#62144)

Fixes interfaces with blank spaces

gamab committed:
Schema: Covering service accounts Core Kind (#62091)

* Adding service accounts schema

Co-authored-by: Eric Leijonmarck <eric.leijonmarck@gmail.com>

* Change metadata field

Co-authored-by: Eric Leijonmarck <eric.leijonmarck@gmail.com>

* Merging the two DTOs into one schema

Co-authored-by: Eric Leijonmarck <eric.leijonmarck@gmail.com>

* Eric dedup things

Co-authored-by: Eric Leijonmarck <eric.leijonmarck@gmail.com>

* Add optional to teams/created/updated

Co-authored-by: Eric Leijonmarck <eric.leijonmarck@gmail.com>

* Lower case the OrgId

Co-authored-by: Eric Leijonmarck <eric.leijonmarck@gmail.com>

Co-authored-by: Eric Leijonmarck <eric.leijonmarck@gmail.com>

Commit:
Tempo: Create separate functions for querying the v1/v2 API for tag values (#61998)

alexweav committed:
Alerting: Copy rule definitions into state history (#62032)

* Copy rules instead of accepting pointer

* Deep-copy the rule, for even more guarantees

* Create struct just for needed fields

* Move RuleMeta to historian/model package, iron out package dependencies

* Move tests for dash ID parsing to model package along with code

mildwonkey committed:
chore: move entity models into entity store service (#62145)

baldm0mma committed:
balm0mma/addNumRows/ build out display method to return numeric and text values

nmarrs committed:
SVG: Add dompurify preprocessor step (#62143)

Co-authored-by: Ryan McKinley <ryantxu@gmail.com>

ryantxu committed:
Preferences: Add preferences kind and remove unused navbar settings (#59621)

ryantxu committed:
k8s: add feature toggle and stub to save dashboards k8s  (#62053)

ryantxu committed:
Chore: Fix dashboard service after refactor (#62153)

merge refactoring sequence

Commit:
Kindsys: Simpler PascalCase name handling (#62150)

alexweav committed:
Alerting: Re-generate swagger definitions (#62154)

Regenerate swagger, add body binding so parameters work

baldm0mma committed:
baldm0mma/addRowNums/ add table showRowNumbers tableOption

baldm0mma committed:
baldm0mma/addRowNums/ add table showRowNumbers tableOption

baldm0mma committed:
baldm0mma/addRowNums/ add table showRowNumbers tableOption

baldm0mma committed:
balm0mma/addRowNums/ remove console.logs and update props

baldm0mma committed:
baldm0mma/addRowNums/ reset default value for addBooleanSwitch

baldm0mma committed:
balm0mma/addRowNums/ make prop optional

baldm0mma committed:
balm0mma/addRowNums/ make prop optional

baldm0mma committed:
baldm0mma/addNumRows/ rename file ext

baldm0mma committed:
baldm0mma/addNumRow/ remove console.log in Table

baldm0mma committed:
balm0mma/addNumRows/ clean Table.tsx

baldm0mma committed:
baldm0mma/addNumRows/ clean and remove superfluous annotations

oscarkilhed committed:
XY Panel: Add and use schema for XY Panel configuration (#62148)

* Add schema for XYChart

* Minor cleanup

* remove unessecary fields from ScatterFieldConfigVeneer, and cleaning up more

* Fix counts

Co-authored-by: sam boyer <sdboyer@grafana.com>

ryantxu committed:
EntityAPI: Rename references from kind to family (#62044)

nmarrs committed:
Canvas: Fix connection anchors for svg elements (#61895)

mdvictor committed:
Refactor state timeline/status history to cue model and refactor `TimelineChart` component (#61631)

* Adapt state timeline to scuemata

* Refactor status history to cue model

* Refactor

* Refactor TimelineChart as a core component

* wip

* Change as per CR

Co-authored-by: sam boyer <sdboyer@grafana.com>

torkelo committed:
Chore: Fixes cleanNeedle regex to be more specific (#61967)

* Chore: Fixes code sec warning for frontend regex

* Update text.ts

* Update text.ts

papagian committed:
Access control: Modify dashboard/folder resolvers so that return also the inherited scopes (#62025)

* Access Control: Add folder service dependency to the dashboard/folder resolvers

* Expose the function fetching parents to folder interface

* Add generic prepend utility

* Modify dashboard resolvers to return inherited scopes

ashharrison90 committed:
Dropdown: Make escape close a dropdown (#62098)

* make escape close a dropdown

* make tab close the dropdown

grobinson-grafana committed:
Alerting: Support context.Context in Loki interface (#61979)

This commit adds support for canceleable contexts in the Loki
interface.

mdvictor committed:
Fix Barchart legend calcs when stacking is percent (#61449)

* Fix Barchart legend calcs when stacking is percent

* doc change

* Refactor + tests

joey-grafana committed:
Tempo: Update column width for Loki search (#61924)

Update col widths

dprokop committed:
Dashboard schema: Review and mature refresh property (#62104)

* Review and mature refresh property

* Update type

* Update report

armandgrillet committed:
Add handleGrafanaManagedAlerts jsonData (#61874)

* Add handleGrafanaManagedAlerts jsonData

* Update following feedback

kalleep committed:
AuthN: Rebuild Authenticate so we only have to call it once in context handler (#61705)

* API: Add reqSignedIn to router groups

* AuthN: Add fall through in context handler

* AuthN: Add IsAnonymous field

* AuthN: add priority to context aware clients

* ContextHandler: Add comment

* AuthN: Add a simple priority queue

* AuthN: Add Name to client interface

* AuthN: register clients with function

* AuthN: update mock and fake to implement interface

* AuthN: rewrite test without reflection

* AuthN: add comment

* AuthN: fix queue insert

* AuthN: rewrite tests

* AuthN: make the queue generic so we can reuse it for hooks

* ContextHandler: Add fixme for auth headers

* AuthN: remove unused variable

* AuthN: use multierror

* AuthN: write proper tests for queue

* AuthN: Add queue item that can store the value and priority

Co-authored-by: Jo <joao.guerreiro@grafana.com>

dsotirakis committed:
Packages: Fix `upload-packages` case for Enterprise2 mode (#61772)

* Add corner case for Enterprise2

* Add versionMode corner case

* Fix lint

* Refactor

* Add tests

* Remove break

* Fix lint

* Return error instead of panicing

* Fix lint

joanlopez committed:
Encryption: Cache new DEKs (only) after a caution period (#60664)

* Encryption: Cache new DEKs (only) after commit

* Fix typo

* Update secrets manager tests with new failing case

* Update secrets manager tests with new clarifications (comments)

* Correct broken method calls

* Unify methods

* Cache data keys only after a caution period

* Caution period for data keys caching only for encrypt ops

ashharrison90 committed:
Navigation: Command palette topnav tweaks (#61991)

stylings tweaks to command palette

torkelo committed:
QueryEditorRow: Fixes issue loading query editor when data source variable selected (#61927)

* QueryEditorRow: Fixes issue loading query editor when data source variable selected

* Fix func name

* Removed wait

* restore waiting for with better check

* restore waiting for with better check

ashharrison90 committed:
Chore: Move to node 18 again (#62079)

move to node 18 again

joshhunt committed:
Chore: Script to generate betterer issue summary (#61857)

* wip

* finish script

* limit by message

* fix filter message

* Add template

* improve template

* codeowners devdep

aangelisc committed:
Bump code-coverage workflow version (#62182)

LudoVio committed:
Transforms: Keep refId for labelsTopFields (#61916)

jackw committed:
Chore: Bump d3-color to 3.1.0 (#61609)

* chore(d3-color): resolve all copies to 3.1.0 to fix vulnerability

* chore(yarn): fix resolution version

* chore(jest): compile d3-color from esm to cjs

* chore(d3): bump d3 to latest, set jest config to transpile es modules

* chore: bump visx packages to resolve d3-color to latest

* chore(jest): add missing es modules to config

* chore(storybook): prevent storybook config failure due to nested esm imports

* chore(storybook): use a direct path to avoid es modules breaking config build

* build(sass): use esbuild to generate sass theme vars for esm deps

* chore: clean up swc dependencies

* revert: remove debug console.log breaking tests

ashharrison90 committed:
Navigation: Add keyboard shortcut to search input (#62116)

* Show shortcut

* add icon to search bar, make same changes in search modal

* rename to modKey

Co-authored-by: joshhunt <josh@trtr.co>

ashharrison90 committed:
Navigation: only show the `img` for a section root if both `img` and `icon` are present (#62127)

only show an img for a section root if both img and icon are present

LudoVio committed:
Transforms: Add join by fields (#61322)

VikaCep committed:
Alerting: validate alert condition on saving rule (#61958)

* Validate alert condition on saving rule

* Remove unused const

kalleep committed:
Auth: Fix error check (#62192)

konrad147 committed:
Alerting: Alert rules search improvements (#61398)

Co-authored-by: Gilles De Mey <gilles.de.mey@gmail.com>

gotjosh committed:
Alerting: Fix Test Receivers when settings are non-strings (#62156)

* Alerting: Fix Test Receivers when settings are non-strings

As part of the Alerting extraction, we want to make sure we don't have circular depedencies. As such, I had to move `PostableGrafanaReceiver` to a new struct in `grafana/alerting` called `GrafanaReceiver`.

`PostableGrafanaReceiver` has an attribute called `Settings` that uses a Grafana-propietary struct called `RawMessage`, this struct shadows `json.RawMessage`.

When I created `GrafanaReceiver`, I turned settings into a `map[string]string` thinking all settings would end up as strings. This was a mistake, and this test proves that it doesn't work, and breaks the API.

renovate[bot] committed:
Update dependency glob to v8.1.0 (#62186)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

leandro-deveikis committed:
Snapshots: Add snapshot enable config (#61587)

* Add config to remove Snapshot functionality (frontend is hidden and validation in the backend)
* Add test cases
* Remove unused mock on the test
* Moving Snapshot config from globar variables to settings.Cfg
* Removing warnings on code

eleijonmarck committed:
Auth: Fix catch both both ErrInvalidAPIKey for context with APIKey (#62193)

* fix: capture both ErrInvalidAPIKey

* rename of variable

mildwonkey committed:
chore: move dashboard_acl models into dashboard service (#62151)

evictorero committed:
Schema: Create PublicDashboard core kind (#62155)

leandro-deveikis committed:
Snapshots: Add new snapshot configuration to documentation (#62110)

* Add new snapshot configuration to documentation
Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

leventebalogh committed:
Navigation: Fix finding the active nav item for plugins (#62123)

fix: look for the active page under children first

ying-jeanne committed:
[Xorm] remove oracle driver + unused function (#62125)

[Xorm] Some more clean up

jdbaldry committed:
Update doc-validator to latest release (#62170)

* Update doc-validator to latest release

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

* Skip image validation

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

IevaVasiljeva committed:
Chore: fix builds on main (#62218)

fix builds

ivanahuckova committed:
Explore: Add feature to open log sample in split view (#62097)

* Add tests

* Implement split open to see logs functionality

* Fix imports in test

* Update packages/grafana-data/src/types/logs.ts

Co-authored-by: Matias Chomicki <matyax@gmail.com>

* Update packages/grafana-data/src/types/logs.ts

Co-authored-by: Matias Chomicki <matyax@gmail.com>

* Update default scneario to throw error

* Exit early in getSupplementaryQuery

* Update public/app/features/explore/LogsSamplePanel.tsx

Co-authored-by: Matias Chomicki <matyax@gmail.com>

soniaAguilarPeiron committed:
Alerting: Fix recording rules being shown in the rules table on edit group modal (#62171)

* Fix recording rules being shown in the rules table on edit group modal

* Show no alerts message when there is not some non-recording rules and add tests

* Move specific mock constants to the test file

Jguer committed:
Server: Remove unused services (#62015)

remove unused entries

Co-authored-by: Gabriel MABILLE <gamab@users.noreply.github.com>

Co-authored-by: Gabriel MABILLE <gamab@users.noreply.github.com>

ivanahuckova committed:
Explore: Trigger logs sample only when user click to see it (#62226)

LogSamples: Trigger log samples on user action

grafanabot committed:
Changelog: Updated changelog for 9.3.6 (#62231)

wbrowne committed:
TestData: Introduce schema types (#62130)

* schematize data query

* add the stuff you dingus

* feat(testdatasource): add scenario to generated types

* use generated testdata query in frontend

* update code owners

* Add path exception for testdata datasource

* use specific numeric data types

* fix test

* fix e2e smoketest

* add test data query type

* use test data query type

* fix betterer

* Fix typo

* move to experimental

Co-authored-by: Alex Khomenko <Clarity-89@users.noreply.github.com>

Co-authored-by: Jack Westbrook <jack.westbrook@gmail.com>
Co-authored-by: Marcus Efraimsson <marcus.efraimsson@gmail.com>
Co-authored-by: sam boyer <sdboyer@grafana.com>
Co-authored-by: Alex Khomenko <Clarity-89@users.noreply.github.com>

dsotirakis committed:
Chore: Update `latest.json` to `9.3.6` (#62235)

Chore: update latest.json to 9.3.6

aocenas committed:
Tempo: Trace to logs custom query with interpolation (#61702)

svennergr committed:
Codeowners: Assign `logs` files to @grafana/observability-logs (#62239)

assign "logs" files to logs squad

ivanahuckova committed:
Explore: Add feature tracking for split open in logs sample (#62241)

Explore: Add tracking for split open in logs sample

gabor committed:
Elasticsearch: Deprecate raw document mode (#62236)

elasticsearch: deprecate raw-document-mode

matyax committed:
Loki Docker Devenv: Fix undefined variable in error message (#62242)

fix(loki-docker-devenv): fix undefined variable in error message

jackw committed:
Fix: NPM Package publishing (#62243)

* move packed packages to npm-artifacts dir

* remove unnecessary any to trigger canary build

* update canary script to create dir only if it does not exist

Co-authored-by: Ashley Harrison <ashley.harrison@grafana.com>

Jguer committed:
AuthN: JWT remove unnecessary if (#62233)

remove unnecessary if

ashharrison90 committed:
Chore: don't create the npm-artifacts dir since yarn packages:pack does this… (#62246)

* don't create the npm-artifacts dir since yarn packages:pack does this already

* fix an any to trigger a canary build

renovate[bot] committed:
Update dependency immutable to v4.2.2 (#62202)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

aocenas committed:
NodeGraph: Schema for panel options (#62121)

baldm0mma committed:
baldm0mma/addRowNums/ update models.gen.ts manually

Commit:
Alerting: Add alert pausing feature (#60734)

* Add field in alert_rule model, add state to alert_instance model, and state to eval

* Remove paused state from eval package

* Skip paused alert rules in scheduler

* Add migration to add is_paused field to alert_rule table

* Convert to postable alerts only if not normal, pernding, or paused

* Handle paused eval results in state manager

* Add Paused state to eval package

* Add paused alerts logic in scheduler

* Skip alert on scheduler

* Remove paused status from eval package

* Apply suggestions from code review

Co-authored-by: George Robinson <george.robinson@grafana.com>

* Remove state

* Rethink schedule and manager for paused alerts

* Change return to continue

* Remove unused var

* Rethink alert pausing

* Paused alerts storing annotations

* Only add one state transition

* Revert boolean method renaming refactor

* Revert take image refactor

* Make registry errors public

* Revert method extraction for getting a folder title

* Revert variable renaming refactor

* Undo unnecessary changes

* Revert changes in test

* Remove IsPause check in PatchPartiLAlertRule function

* Use SetNormal to set state

* Fix text by returning to old behaviour on alert rule deletion

* Add test in schedule_unit_test.go to test ticks with paused alerts

* Add coment to clarify usage of context.Background()

* Add comment to clarify resetStateByRuleUID method usage

* Move rule get to a more limited scope

* Update pkg/services/ngalert/schedule/schedule.go

Co-authored-by: George Robinson <george.robinson@grafana.com>

* rum gofmt on pkg/services/ngalert/schedule/schedule.go

* Remove defer cancel for context

* Update pkg/services/ngalert/models/instance_test.go

Co-authored-by: Santiago <santiagohernandez.1997@gmail.com>

* Update pkg/services/ngalert/models/testing.go

Co-authored-by: Santiago <santiagohernandez.1997@gmail.com>

* Update pkg/services/ngalert/schedule/schedule_unit_test.go

Co-authored-by: Santiago <santiagohernandez.1997@gmail.com>

* Update pkg/services/ngalert/schedule/schedule_unit_test.go

Co-authored-by: Santiago <santiagohernandez.1997@gmail.com>

* Update pkg/services/ngalert/models/instance_test.go

Co-authored-by: Santiago <santiagohernandez.1997@gmail.com>

* skip scheduler rule state clean up on paused alert rule

* Update pkg/services/ngalert/schedule/schedule.go

Co-authored-by: Santiago <santiagohernandez.1997@gmail.com>

* Fix mock in test

* Add (hopefully) final suggestions

* Use error channel from recordAnnotationsSync to cancel context

* Run make gen-cue

* Place pause alert check in channel update after version check

* Reduce branching un update channel select

* Add if for error and move code inside if in state manager ResetStateByRuleUID

* Add reason to logs

* Update pkg/services/ngalert/schedule/schedule.go

Co-authored-by: George Robinson <george.robinson@grafana.com>

* Do not delete alert rule routine, just exit on eval if is paused

* Reduce branching and create-close a channel to avoid deadlocks

* Separate state deletion and state reset (includes history saving)

* Add current pause state in rule route in scheduler

* Split clearState and bring errCh closer to RecordStatesAsync call

* Change rule to ruleMeta in RecordStatesAsync

* copy state to be able to modify it

* Add timeout to context creation

* Shorten the timeout

* Use resetState is rule is paused and deleteState if rule is not paused

* Remove Empty state reason

* Save every rule change in historian

* Add tests for DeleteStateByRuleUID and ResetStateByRuleUID

* Remove useless line

* Remove outdated comment

Co-authored-by: George Robinson <george.robinson@grafana.com>
Co-authored-by: Santiago <santiagohernandez.1997@gmail.com>
Co-authored-by: Armand Grillet <2117580+armandgrillet@users.noreply.github.com>

baldm0mma committed:
baldm0mma/addRowNums/ remove double prop

baldm0mma committed:
baldm0mma/addRowNums/ update default options

baldm0mma committed:
baldm0mma/addRowNums/ remove annotation for chore

baldm0mma commented:
Closing this pull request and opening a similar one to follow the BI Squad conventions on PRs
